### PR TITLE
test metafeed announcement & resulting hops

### PR DESCRIPTION
**Pull request by cblgh**
related to #115

### CODE REVIEW

**Comment by cblgh** [comment]
status: currently running into these two compilation errors in my implementation sketch:

```
./builder_indexing.go:53:10: c.UnmarshalJSON undefined (type metafeedAnnounceMsg has no field or method UnmarshalJSON)
./builder_indexing.go:61:27: cannot use c.Metafeed (type string) as type refs.FeedRef in argument to storedrefs.Feed
```

they are kind of related to the comments i've sprinkled around already. the first error will be fixed by using the already defined `metafeedAnnouncement` struct (iirc: i'm not using it because it doesn't have the uri bits?), and the second by waiting for the [ssb-uri-flavoured refactor](https://github.com/cryptoscope/ssb/pull/116) to land

File: graph/builder_indexing.go [hunk 1]
```diff
@@ -29,6 +29,46 @@ const (
...
+	err := c.UnmarshalJSON(abs.ContentBytes())
+	if err != nil {
+		// just ignore invalid messages, nothing to do with them (unless you are debugging something)
+		//level.Warn(b.log).Log("msg", "skipped contact message", "reason", err)
+		return nil
+	}
+
+	addr := storedrefs.Feed(abs.Author())
+	addr += storedrefs.Feed(c.Metafeed)
+	err = idx.Set(ctx, addr, idxRelValueFollowing)
```
Comment (cryptix) [inline graph/builder_indexing.go:62]:
```suggestion
	err = idx.Set(ctx, addr, idxRelValueMetafeed)
```

File: graph/metafeeds_test.go [hunk 2]
```diff
@@ -83,6 +85,134 @@ var metafeedsScenarios = []PeopleTestCase{
...
+		// name: "followed peer has newly announced metafeed",
+	{
+		name: "follow peer with metafeed announcement",
+		ops: []PeopleOp{
+			PeopleOpNewPeer{"alice-main"},
+			PeopleOpNewPeer{"bob-main"},
+			PeopleOpFollow{"alice-main", "bob-main"},
+			PeopleOpNewPeerWithAlgo{"bob-mf", refs.RefAlgoFeedBendyButt},
+			PeopleOpAnnounceMetafeed{"bob-main", "bob-mf"},
+			// question: do we need to add a new op for adding an existing subfeed?
```
Comment (cryptix) [inline graph/metafeeds_test.go:99]: nope, this is fine.

there should be a seperate testcase for AddExisting on the metafeed.

File: graph/metafeeds_test.go [hunk 3]
```diff
@@ -83,6 +85,134 @@ var metafeedsScenarios = []PeopleTestCase{
...
+			// i.e. PeopleOpExistingSubFeed, instead of PeopleOpNewSubFeed
+
+			// question: how do we actually test that the announcement works correctly?
+			// i assume we need to create a normal feed (normie) following bob-main, and then check that 
+			// the hops from normie->{bob-mf,bob-main,bob-indexes: about} == 0 (using hops 0 according to go-ssb hops conventions)
+			//
+			// following that logic: we should also have a test that maliciously creates an announcement for a metafeed that
```
Comment (cryptix) [inline graph/metafeeds_test.go:106]: I'm fairly certain we don't need this.

TODO: make this more clear in the the metafeed spec?

File: graph/metafeeds_test.go [hunk 4]
```diff
@@ -83,6 +85,134 @@ var metafeedsScenarios = []PeopleTestCase{
...
+
+	// mfAddExisting <=> "metafeed/add/existing"
+	mfAddExisting := metamngmt.NewAddExistingMessage(kpMetafeed.ID(), kpMain.ID(), "main")
+	mfAddExisting.Tangles["metafeed"] = refs.TanglePoint{Root: nil, Previous: nil}
+
+	signedAddExistingContent, err := metafeed.SubSignContent(kpMain.Secret(), mfAddExisting)
+	if err != nil {
+		return err
+	}
+	mf.publish.Append(signedAddExistingContent)
```
Comment (cryptix) [inline graph/metafeeds_test.go:213]: This should be a separate Op

**Review by cryptix — COMMENTED** [review]
good first draft!

(reviewed via call)

File: graph/metafeeds_test.go (hunk 4 above)
Comment (cblgh) [reply graph/metafeeds_test.go:213]: done! :+1:

**Review by cryptix — COMMENTED** [review]
SGTM but let's hold off with merging until #116 is in.

**Pull request merged**